src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -475,6 +487,14 @@ internal static List<CompletionResult> CompleteModuleName(CompletionContext cont
                         && completionText.Contains('.')
                         && !shortNamePattern.IsMatch(completionText.Substring(completionText.LastIndexOf('.') + 1))
                         && !shortNamePattern.IsMatch(completionText))
+                    {
+                        // This check is to make sure we don't return a module whose name only matches the user specified word in the middle.
+                        // For example, when user completes with 'gmo power', we should not return 'Microsoft.PowerShell.Utility'.
+                        continue;
```

Comment:
Do you want to revert https://github.com/PowerShell/PowerShell/pull/20330?

/cc @MartinGC94